src/chainstate/stacks/db/mod.rs

```diff
@@ -1001,6 +1030,154 @@ impl StacksChainState {
                 });
             }
 
+            let bns_contract_id = boot_code_id("bns");
+            if let Some(get_namespaces) = boot_data.get_bulk_initial_namespaces.take() {
+                info!("Initializing chain with namespaces");
+                clarity_tx.connection().as_transaction(|clarity| {
+                    clarity
+                        .with_clarity_db(|db| {
+                            let initial_namespaces = get_namespaces();
+                            for entry in initial_namespaces {
+                                let namespace = {
+                                    let buffer = entry.namespace_id.as_bytes();
+                                    Value::buff_from(buffer.to_vec()).expect("Invalid namespace")
+                                };
+
+                                let importer = {
+                                    let address = StacksChainState::parse_genesis_address(
+                                        &entry.importer,
+                                        mainnet,
+                                    );
+                                    Value::Principal(address)
+                                };
+
+                                let revealed_at = Value::UInt(entry.revealed_at.into());
+                                let launched_at = Value::UInt(entry.launched_at.into());
+                                let lifetime = Value::UInt(entry.lifetime.into());
+                                let price_function = {
+                                    let base = Value::UInt(entry.base.into());
+                                    let coeff = Value::UInt(entry.coeff.into());
+                                    let nonalpha_discount =
+                                        Value::UInt(entry.nonalpha_discount.into());
+                                    let no_vowel_discount =
+                                        Value::UInt(entry.no_vowel_discount.into());
+                                    let buckets: Vec<_> = entry
```

Comment:
You may want to verify that there are 16 entries here.

Reply:
Checks added in https://github.com/blockstack/stacks-blockchain/pull/2204/commits/52579d5dd4d88987a9dbc53543c8bb85ae971cbc